### ISSUE TITLE
Navigate by date in Timeline view

### ISSUE THREAD

**Issue by viktorstrate** [photoview/photoview#295]
_This is a continuation of issue #48_

> ### Suggestion / inspiration for timeline view
> ![Home](https://user-images.githubusercontent.com/4233458/100153151-05b20880-2ea4-11eb-87ae-a22ad246036a.png)

I may miss something but I do not see the timeline at right inspired by google photos...
Is there an option? or may-be I have to wait the full scan process to finish?

thanks

_Originally posted by @darrepac in https://github.com/photoview/photoview/issues/48#issuecomment-813029187_

---

The seemless timeline feature of Google Photos, is very complicated to implement whilst still being performant.
The best option would be to rethink how to navigate by dates, maybe it could still have the years/dates in the side, but when a date is clicked it would refetch starting from that date and forwards.

I'd like to hear some oppinions/suggestions to this.

See this article about what goes into the timeline feature of Google Photos
https://medium.com/google-design/google-photos-45b714dfbed1

**Comment by darrepac** [comment]
For information, Plex has a timeline view as Google Photo:
![image](https://user-images.githubusercontent.com/7128994/113515341-33a30a00-9574-11eb-9886-21222cf9a7f7.png)

**Comment by MinchinWeb** [comment]
Yes, this would be wonderful. I'd love to be able to jump back in time, and scrolling through 10 years of photos (and waiting for *all* the thumbnails to load) isn't particularly workable...

**Comment by darrepac** [comment]
Is there any news on this feature request?

**Comment by viktorstrate** [comment]
@darrepac I have been delaying this as I didn't know how to do it.

For the redesign mockup, I've experimented with the idea of adding a date filter to show images from a certain date.
I'd like to hear your thoughts on that

![](https://user-images.githubusercontent.com/4233458/119526032-4f11e080-bd7f-11eb-9562-bdc38be79747.jpg)

**Comment by darrepac** [comment]
Summer is behind us. You had to relax, refresh your mind and... find ideas on how to implement that! :D
Many self hosted open source google photo replacement...but none with a working timeline infinite view. Everyone is looking at it, the one that will make it will win! ;)

**Comment by viktorstrate** [comment]
@darrepac don't get your expectations too high, what google has managed with Google Photos is quite impressive.
However, I am working on rewriting the timeline to be simpler and more flexible, both for the API and the UI. (see also https://github.com/photoview/photoview/issues/230#issuecomment-913716684).

But I hope that, some day in the future, I will be able to pull this off. And the upcoming changes to the timeline will for sure be a step in the right direction.

----

> Summer is behind us. You had to relax, refresh your mind and... find ideas on how to implement that!

I know you probably don't mean it like that, but remember that I am mostly working on Photoview alone and in my spare time and for my own sake. You can not expect me to just solve this, just because a couple of months has gone by.

Also I'd like to mention that I have been quite busy the last couple of months, working on a release that fixed quite a lot of bugs. I think one can easily forget how long it takes to maintain and fix bugs, and think that nothing is happening just because no new features are added.

**Comment by darrepac** [comment]
Yes, there was no bad intentions in my comment, just emphasize from what I read here and there that infinite timeline view is a must... But sure, the amount of works already achieved is impressive and the work to maintain is not to be underestimated too! Keep going!

**Comment by viktorstrate** [comment]
Cool, I'm glad we're on the same page then :blush: I agree that the timeline sucks at the moment. But the first step will be to show all images, no more +xx images, and have a select date dropdown option. 

Then later when I feel for a big challenge I can try to implement a Google Photos like scroll to date feature.

**Comment by egnappahz** [comment]
The beauty of photoview remains that it just represents the filesystem (via albums) as it is.

So, I created a simple crontab'd bash script that creates symlinks per year, and per month (judging by filestats, filemetastats, or filename of the image, until I get a correct date) in organised folders. I now have a, separate, organised "timeline-view" the way I want it (monthly seperation) without going into the PHP code and I think its wonderful.

Thank you for all your work!
